### ISSUE TITLE
Should update file's stat.mtime

### ISSUE THREAD

**Issue by damonmaria** [lazd/gulp-replace#88]
If gulp-replace is used in a stream and modifies a file, then `gulp.dest()` overwrites a previous output then a lot of tools (e.g. `rsync` by default) will not realise the file is changed because it will still have the same size and mtime. 

IMHO if gulp-replace modifies a Vinyl file in the stream then it should also set `stat.mtime` to now.

**Comment by lazd** [comment]
Seems like vinyl should do that on its own... Perhaps it's related to https://github.com/gulpjs/vinyl-fs/issues/124 ?

**Comment by damonmaria** [comment]
I tried to follow the various Vinyl issues about this. I think in the end they decided against updating `mtime` when content is changed. But it's pretty hard to untangle all the different discussions.

**Comment by lazd** [comment]
@phated what is best practice here? Is this the plugin's responsibility or vinyl's?

**Comment by phated** [comment]
@lazd this is a known limitation of newer vinyl-fs versions that can't be solved until we implement https://github.com/gulpjs/vinyl/issues/105 - It's recommended for end-users to implement a `touch`-like stream if they are relying on the filesystem `mtime`.  As always, it's __not__ recommended for plugin authors to try to catch-all everyone's use case and should instead defer to pluggable pieces.

Example:
```js
var touch = through.obj(function(file, enc, cb){
  file.stat.mtime = new Date();
  cb(null, file);
});
```

**Comment by lazd** [comment]
Thanks for the super quick reply, @phated!

@damonmaria It looks like we don't need to do this as part of `gulp-replace`, so I'm going to close this issue.

**Comment by damonmaria** [comment]
That is exactly what I ended up doing. But then found it strange that there's no existing gulp plugin to do this. The two 'touch' plugins I found both seem designed to modify the file after `gulp.dest()`. 

I understand the issues with modifying mtime based on a content update but it does lead to confusing behaviour that may not show up immediately. I had both `rsync` and `make` missing changes because of this.
